packages/eas-cli/src/user/SessionManager.ts

```diff
@@ -229,6 +271,19 @@ export default class SessionManager {
     });
   }
 
+  private async sudoLoginAsync(input: { password: string; otp?: string }): Promise<void> {
+    const authenticationInfo = {
+      accessToken: this.getAccessToken(),
+      sessionSecret: this.getSessionSecret(),
+    };
+    const apiV2Client = new ApiV2Client(authenticationInfo);
+    await upgradeSudoSessionAndAsync(apiV2Client, input);
+  }
+
+  private async ssoSudoUpgradeAsync(): Promise<void> {
+    throw Error('Sudo upgrade with SSO is not yet supported');
+  }
```

Comment:
Yep. This will need to do something similar to how SSO login is done in the CLI (using the website I believe). Can trace the SSO code in the CLI `ssoLoginAsync` to see how SSO auth works. Happy to dig deeper if necessary.

Reply:
Would be good to stack a PR on top of this that adds SSO and then land them together.